app/services/publisher_login_link_emailer.rb

```diff
@@ -0,0 +1,65 @@
+# "Magic sign in link" / One time sign-in token via email
+class PublisherLoginLinkEmailer
+  attr_accessor :error
+  attr_reader :brave_publisher_id, :normal_publisher_id, :email, :publisher
+
+  def initialize(brave_publisher_id:, email:)
+    @brave_publisher_id = brave_publisher_id
+    @email = email.presence
+  end
+
+  # Returns true if everything worked;
+  # Returns false and sets #error if something didn't work.
+  def perform
+    normalize_brave_publisher_id \
+      && find_publisher \
+      && send_email
+  end
+
+  def normalize_brave_publisher_id
+    require "faraday"
+    @normal_publisher_id = PublisherDomainNormalizer.new(
+      domain: brave_publisher_id
+    ).perform
+  rescue PublisherDomainNormalizer::DomainExclusionError, Faraday::Error
+    @error = I18n.t("activerecord.errors.models.publisher.attributes.brave_publisher_id.api_error_cant_normalize")
+    false
+  end
+
+  def find_publisher
+    publisher_verified = Publisher.find_by(brave_publisher_id: brave_publisher_id, verified: true)
+    if publisher_verified
+      # For verified publishers, email can be blank.
+      if !email || publisher_verified.email == email
+        @publisher = publisher_verified
+        return true
+      else
+        @error = I18n.t("publishers.new_auth_token_wrong_email_publisher_verified")
+        return false
+      end
+    end
+
+    publishers_not_verified = Publisher.where(
+      brave_publisher_id: brave_publisher_id,
+      verified: false
+    )
+    if publishers_not_verified.none?
+      @error = I18n.t("publishers.new_auth_token_publisher_not_found")
```

Comment:
This is a good thought; I considered this but decided it was okay because browser-laptop shows the verified icon next to domains. It's also simple to creep on domains there: open a new profile, enable Payments, turn down min visit criteria, then browse a few websites.
@mrose17 @BrendanEich